symbolic-debuginfo/src/sourcebundle.rs

```diff
@@ -891,18 +903,30 @@ impl<'data> SourceBundleDebugSession<'data> {
     }
 
     /// Looks up a source file descriptor.
+    ///
+    /// The file is looked up in both the embedded files and
+    /// in the included source link mappings, in that order.
     fn get_source_file_descriptor(
         &self,
         key: FileKey,
     ) -> Result<Option<SourceFileDescriptor<'_>>, SourceBundleError> {
-        let zip_path = match self.index.indexed_files.get(&key) {
-            Some(zip_path) => zip_path.as_str(),
-            None => return Ok(None),
-        };
-
-        let content = self.source_by_zip_path(zip_path)?;
-        let info = self.index.manifest.files.get(zip_path);
-        Ok(content.map(|opt| SourceFileDescriptor::new_embedded(Cow::Owned(opt), info)))
+        match self.index.indexed_files.get(&key) {
+            Some(zip_path) => {
+                let zip_path = zip_path.as_str();
+                let content = self.source_by_zip_path(zip_path)?;
+                let info = self.index.manifest.files.get(zip_path);
+                Ok(content.map(|opt| SourceFileDescriptor::new_embedded(Cow::Owned(opt), info)))
+            }
+            None => {
+                let FileKey::Path(path) = key else {
+                    return Ok(None);
+                };
```

Comment:
fun fact, now that we have `rustfmt` support for let-else, if you remove the `;` from the inner return, it will keep auto-formating things onto a single line, if it was on a single line to begin with.